orleans/Adventure/AdventureGrains/PlayerGrain.cs

```diff
@@ -26,8 +26,13 @@ public override Task OnActivateAsync(CancellationToken cancellationToken)
     async Task IPlayerGrain.Die()
     {
         // Drop everything
-        var tasks = _things.Select(Drop).ToList();
-        await Task.WhenAll(tasks);
+        var dropTasks = new List<Task>();
+        var copy = new List<Thing>(_things);
+        foreach (var thing in copy)
+        {
+            dropTasks.Add(this.Drop(thing));
+        }
+        await Task.WhenAll(dropTasks);
```

Comment:
```suggestion
        var dropTasks = new List<Task<string?>>();        
        foreach (var thing in _things.ToArray() /* New collection */)
        {
            dropTasks.Add(Drop(thing));
        }
        await Task.WhenAll(dropTasks);
```

Let's do this instead.